ddl/ddl_api.go

```diff
@@ -3364,7 +3344,7 @@ func CheckModifyTypeCompatible(origin *types.FieldType, to *types.FieldType) (al
 	originUnsigned := mysql.HasUnsignedFlag(origin.Flag)
 	if originUnsigned != toUnsigned {
 		msg := fmt.Sprintf("can't change unsigned integer to signed or vice versa")
-		if isIntType {
+		if canChange {
```

Comment:
`canChange` is different with `isIntType`. Even if `to.Type` is not int, it still reports integer type change error.

Reply:
It's judge `originUnsigned != toUnsigned` for integer.